python/sglang/lang/interpreter.py

```diff
@@ -724,8 +727,44 @@ def _execute_concatenate_and_append_kv_cache(self, expr: SglConcateAndAppend):
         src_rids = [state.stream_executor.sid for state in expr.states]
         self.backend.concatenate_and_append(src_rids, self.sid)
 
+    def _execute_separate_reasoning(self, expr: SglSeparateReasoning):
+        if self.stream:
+            # separate reasoning for stream is not supported
+            return
```

Comment:
Shall we raise warnings when users use `seperate_reasoning` in stream mode?

Reply:
I debated about it, the idea is to not affect original behavior while also be flexible, I see such pattern a couple of times in sglang code base